### PR TITLE
SF-1326: Expose the management of profiles

### PR DESCRIPTION
**Pull request by rklind**

### CODE REVIEW

File: src/commands/profiles/add.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+			} ,
+			'templates': {
+				desc: 'A comma separated list of the job templates to associate to this new profile.  Both the ID and kind of the job template must be specified (ie. 123456:transfer).',
+				type: 'string'
+			}
+		});
+	},
+	handler: argv => {
+		runCommand(argv, async (client, output) => {
+			let newProfileRequestBody = {
```
Comment (smbecker) [inline src/commands/profiles/add.ts]: `const`

File: src/commands/profiles/download.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+				type: 'string',
+				default: (process.cwd())
+			}
+		})
+	},
+	handler: argv => {
+		runCommand(argv, async (client, output) => {
+			try {
+				const downloader = new FileDownloader(client.httpClient, client.profiles);
+				let path = await downloader.download(argv.id, argv.targetDirectory);
```
Comment (smbecker) [inline src/commands/profiles/download.ts]: `const`

File: src/commands/profiles/add.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+function isJobTemplateKindValid(jobTemplateKind) {
+	if (jobTemplateKinds.indexOf(jobTemplateKind) > -1) {
+		return true
+	}
+	return false;
+}
+
+function getJobTemplatesJson(argv, output) {
+	return argv.templates && argv.templates.split(',').map(input => {
+		const jobTemplateInputParsed = input.trim().split(':');
```
Comment (smbecker) [inline src/commands/profiles/add.ts]: Default to 'transfer' type, which will allow you to use destructuring...

`const [jobTemplateId, jobTemplateKind = 'transfer'] = input.trim().split(':');`

Comment (rklind) [reply src/commands/profiles/add.ts]: will do!

File: src/commands/profiles/add.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+]
+
+function isJobTemplateKindValid(jobTemplateKind) {
+	if (jobTemplateKinds.indexOf(jobTemplateKind) > -1) {
+		return true
+	}
+	return false;
+}
+
+function getJobTemplatesJson(argv, output) {
```
Comment (smbecker) [inline src/commands/profiles/add.ts]: Sorry, last thing... `output` is no longer used here.

Comment (rklind) [reply src/commands/profiles/add.ts]: will do, thanks!

**Pull request merged**